initial-push-jumpstart-part-one (#245)
ippontech/blog-usa: posts/jumpstarting-your-data-engineering-career:-a-beginner's-guide-to-the-data-stack-part-one-fivetran.md

The data stack is a collection of tools and technologies that work together to enable efficient and scalable management of data. In this series, we will review a stack organized around four key components: ingest, transform, load, and orchestrate. The first step in the data stack is the ingest phase, where data is pulled from various sources and made available for further processing. This is where tools like Fivetran come into play. Once the data has been ingested, the next step is to transform it into a usable format for analysis. This is where we'll cover tools like dbt. Once the data has been transformed, the next step is to load it into a data warehouse, like Snowflake. Finally, the orchestrate phase involves scheduling and automating the data pipeline, and is where we'll discuss tools like Airflow. Overall, the data stack is a powerful set of tools that provides a comprehensive framework for managing data at scale, from ingestion to analysis. By leveraging these tools, data professionals and engineers alike can create efficient, reliable, and scalable data infrastructures that can help businesses thrive in the rapidly evolving world of data.

## Ingestion

> Ingestion, the first component in our stack, refers to the process of extracting data from various sources and bringing it together for further processing. Ingestion tools can enable data engineers to quickly and easily connect to a variety of sources, such as databases, SaaS applications, cloud storage services, and other APIs, and synchronize the data into a single space. This process can include several key steps, including configuring connection to the data source, defining the data to be extracted, and scheduling the data sync freqency. Ingestion tools provide an advantage here through their user interfaces, simplifying the data integration process and eliminating the need for manual data ingestion, whcih can be time-consuming and error-prone. By leveraging ingestion tools, data engineers can ensure that data is available and ready for processing. Additionally, ingestion tools can help to maintain data accuracy, completeness, and consistency, ensuring that data is reliable and trustworthy for businesseses to transform in their use cases. Overall, the ingestion phase of the data stack is critical for estalishing a reliable and efficient data infrastructure from square one, enabling businesses to easily extract and integrate data from where it can be best used. 

Reviewer: ```suggestion
Ingestion, the first component in our stack, refers to the process of extracting data from various sources and bringing it together for further processing. Ingestion tools can enable data engineers to quickly and easily connect to a variety of sources, such as databases, SaaS applications, cloud storage services, and other APIs, and synchronize the data into a single space. This process can include several key steps, including configuring a connection to the data source, defining the data to be extracted, and scheduling the data sync frequency. Ingestion tools provide an advantage here through their user interfaces, simplifying the data integration process and eliminating the need for manual data ingestion, which can be time-consuming and error-prone. By leveraging ingestion tools, data engineers can ensure that data is available and ready for processing. Additionally, ingestion tools can help to maintain data accuracy, completeness, and consistency, ensuring that data is reliable and trustworthy for businesses to transform for their use cases. Overall, the ingestion phase of the data stack is critical for establishing a reliable and efficient data infrastructure from square one, enabling businesses to easily extract and integrate data from where it can be best used. 
```

Pros:
- Easy to set up and use
    - Designed to be easy to use, it requires minimal technical knowhow
    - A user friendly interface allows quick connection and data load
- Wide range of integrations
    - Supports over 150 pre-built connectors to sources such as databases and marketing tools
    - Easy connection, no need to design a data stack with pre existing constraints 
- Automated data ingestion
    - Automatically fetches and loads data from sources on a predefined schedule
    - Ensures data is up to date and ready for future analysis
- Real time monitoring
    - Provides real time monitoring and alerts
    - Facilitates smooth data flow

Cons:
- Limited transformation capabilities
    - Primarily designed for ingestion, so its transformation capabilities are limited

>     - Users will likely need other tools to transform and clan data before loading into storage

Reviewer: ```suggestion
    - Users will likely need other tools to transform and clean data before loading into storage
```

- Limited flexibility
    - While it can support a wide range of integrations, it may not be able to support every custom integration that an organization needs
- Cost
    - Charges based on the number of data sources and the amount of ingested data

>     - May be a challenge for organizations with a large number of data sources of a high volume of data 

Reviewer: ```suggestion
    - May be a challenge for organizations with a large number of data sources of high volume of data 
```

2. Connect your source to ingest from.

We can now connect to our ingestion source. We can do this within Fivetran's interface, which offers many connector types to technolgoies like S3, GCP, Instagram Business, and more. 

![Ingestion Source Options](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2023/02/source-types.png)

After selecting the correct connector type, completing the required fields, and following along with Fivetran's easy-to-understand setup guides if need be, we can select 'save and test'. Here is an example of what a connector would like look, in this case S3.

![Ingestion Source Options](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2023/02/example-s3-connection.png)

If the tests are successful, ingestion can begin.

![Ingestion Source Options](https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2023/02/connection-tests.png)

3. Sync'd

Depending on the data's source and volume, it may take a moment for syncing to complete. Once finished, the ingested data is now ready and can be processed for further use in the next stage if the data stack!

## Conclusion

> We’ve discussed the basics of the data stack, the ingestion phase, and the role Fivetran can play within a businesses data infrastructure. In the upcoming post in this series we'll progress to the transformation phase. 

Author: Thanks for reviewing!

Cons:
- Limited transformation capabilities

>     - Primarily designed for ingestion, so its transformation capabilities are limited

Reviewer 2: Might be worth adding a little context here, as Fivetran has essentially deprecated SQL transformations and push people to leverage dbt now.

https://fivetran.com/docs/transformations/dbt

Author: Thanks for the help with wording this nicely!
